src/Storages/StorageLog.cpp

```diff
@@ -192,8 +193,17 @@ Chunk LogSource::generate()
     /// How many rows to read for the next block.
     size_t max_rows_to_read = std::min(block_size, rows_limit - rows_read);
     std::unordered_map<String, ISerialization::SubstreamsCache> caches;
+    std::unordered_map<String, ISerialization::SubstreamsDeserializeStatesCache> deserialize_states_caches;
```

Comment:
@Avogar where are we using this (`deserialize_states_caches`) map ?

Reply:
On line 203 in `readPrefix`. It's needed to read subcolumns that share common prefixes correctly